tic-tac-toe/tictactoe.py

```diff
@@ -7,23 +7,69 @@ class Play(object):
 
     def __str__(self):
-        return self.game[0].__str__() + '\n' + self.game[1].__str__() + "\n" + self.game[2].__str__()
+        return "\n".join(map(str, self.game))
 
+    def check_finished(self):
+        result = False
+        for i in range(2):
+            if self.game[i][0] == self.game[i][1] == self.game[i][2] != '-':
+                result = self.game[i][0]
+            elif self.game[0][i] == self.game[1][i] == self.game[2][i] != '-':
+                result = self.game[i][0]
+        if self.game[0][0] == self.game[1][1] == self.game[2][2] != '-':
+            return self.game[0][0]
+        elif self.game[0][2] == self.game[1][1] == self.game[2][0] != '-':
+            return self.game[0][2]
+        elif map(str, self.game).__contains__("-"):
+            return "Draw"
+        else:
+            return result
 
-def make_move(play, sign, x, y):
-    if int(x) > 2 or int(y) > 2 or int(x) < 0 or int(y) < 0:
-        raise EnvironmentError(" Arguments greater then 2 or less then 0 are not allowed")
-    play.game[x][y] = sign
-    play.move_count += 1
-    print play
 
+    def make_move(self, sign, x, y):
+        self.game[x][y] = sign
+        self.move_count += 1
+        self.finished = self.check_finished()
+        print self
 
-play = Play()
+    def check_satisfaction(self, x, y):
+        try:
+            x, y = int(x), int(y)
+        except ValueError:
+            return "Please enter integers, try again"
+        if not (0 <= x <= 2 and 0 <= y <= 2):
+            return "Arguments greater then 2 or less then 0 are not allowed, try again"
+        if self.game[x][y] != '-':
+            return "Field has been already occupied, try again"
+        else:
+            return "Ok"
 
-while not play.finished:
-    if play.move_count%2 == 1:
-        print ("First player, please")
-        make_move(play, '0', input("insert x -> "), input("insert y -> "))
+
+def main():
+    play = Play()
+    while not play.finished:
+        if play.move_count % 2 == 1:
+            print ("First player, please")
+            sign = '+'
+        else:
+            print ("Second player, please")
+            sign = '0'
+
+        x, y = raw_input("insert x -> "), raw_input("insert y -> ")
+        satisfactory = play.check_satisfaction(x, y)
+        while satisfactory is not "Ok":
+            print satisfactory
+            x, y = raw_input("insert x -> "), raw_input("insert y -> ")
+            satisfactory = play.check_satisfaction(x, y)
+        play.make_move(sign, int(x), int(y))
     else:
-        print ("Second player, please")
-        make_move(play, '+', input("insert x -> "), input("insert y -> "))
\ No newline at end of file
+        if play.finished == '+':
+            print "First player has won!"
+        elif play.finished == '0':
+            print "Second player has won!"
+        elif play.finished == 'Draw':
+            print "The result is draw!"
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
How about adding some [unit tests](https://docs.python.org/2/library/unittest.html) for the code? This will probalby require some restructuring.